src/ion/process.rs

```diff
@@ -728,6 +739,221 @@ impl<'a, F: Function> Env<'a, F> {
         }
     }
 
+    /// Splits the given bundle into minimal bundles per Use, falling
+    /// back onto the spill bundle. This must work for any bundle no
+    /// matter how many conflicts.
+    pub fn split_into_minimal_bundles(&mut self, bundle: LiveBundleIndex, reg_hint: PReg) {
+        let mut removed_lrs: FxHashSet<LiveRangeIndex> = FxHashSet::default();
+        let mut removed_lrs_vregs: FxHashSet<VRegIndex> = FxHashSet::default();
+        let mut new_lrs: SmallVec<[(VRegIndex, LiveRangeIndex); 16]> = smallvec![];
+        let mut new_bundles: SmallVec<[LiveBundleIndex; 16]> = smallvec![];
+
+        let spill = self
+            .get_or_create_spill_bundle(bundle, /* create_if_absent = */ true)
+            .unwrap();
+
+        trace!(
+            "Splitting bundle {:?} into minimal bundles with reg hint {}",
+            bundle,
+            reg_hint
+        );
+
+        let mut last_lr: Option<LiveRangeIndex> = None;
+        let mut last_bundle: Option<LiveBundleIndex> = None;
+        let mut last_inst: Option<Inst> = None;
+        let mut last_vreg: Option<VRegIndex> = None;
+
+        for entry_idx in 0..self.bundles[bundle.index()].ranges.len() {
```

Comment:
Random stylistic comment but you can `impl<T> Index<MyLocalType> for [T]` which could help change all the indexing here to `self.bundles[bundle]` (or even `self[bundle]`) rather than `self.bundles[bundle.index()]`

Reply:
I briefly played with this but it seems we would need wrapper types around the `Vec<T>` to make this work properly. I'd love to have this eventually but it's a bigger refactor (all use-sites across the allocator) than I want to bite off right now; I'll file an issue!

Reply:
#62